contracts/ScoutGameProtocol.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.26;
+
+import "@ethereum-attestation-service/eas-contracts/contracts/resolver/SchemaResolver.sol";
+import "@ethereum-attestation-service/eas-contracts/contracts/IEAS.sol";
+import "./ScoutToken.sol";
+
+/// @title ScoutGameProtocol
+/// @notice A schema resolver that manages unclaimed balances based on EAS attestations.
+contract ScoutGameProtocol is SchemaResolver {
+    ScoutToken public token;
+    mapping(address => uint256) private _unclaimedBalances;
+
+    // Define a constant for 18 decimals
+    uint256 constant DECIMALS = 10 ** 18;
+
+    constructor(IEAS eas, address _token) SchemaResolver(eas) {
+        token = ScoutToken(_token);
+    }
+
+    // Method that is called by the EAS contract when an attestation is made
+    //
+    function onAttest(
+        Attestation calldata attestation,
+        uint256 /*value*/
+    ) internal override returns (bool) {
+        uint256 addedBalance = 10 * DECIMALS;
+        _unclaimedBalances[attestation.recipient] += addedBalance;
+        return true;
+    }
+
+    // Method that is called by the EAS contract when an attestation is revoked
+    function onRevoke(
+        Attestation calldata attestation,
+        uint256 /*value*/
+    ) internal pure override returns (bool) {
+        return true;
+    }
+
+    function getUnclaimedBalance(
+        address account
+    ) public view returns (uint256) {
+        return _unclaimedBalances[account];
+    }
+
+    function getTokenBalance(address account) public view returns (uint256) {
+        return token.balanceOf(account);
+    }
+
+    // Allow the sender to claim their balance as ERC20 tokens
+    function claimBalance(uint256 amount) public returns (bool) {
```

Comment:
Why add an amount here? We could just make it claimBalance, and zero out the user's balance